### PR TITLE
PluginsCatalog: adding error information about disabled plugins.

### LINKED ISSUE #38719: Plugin Catalog: Handle loading errors
The Plugin Catalog doesn't handle various error conditions like if the gcom api isn't accessible or a plugin.json file doesn't load correctly.

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
If any of the plugins, for some reason, fails during startup we will indicate that in the plugin catalog and adding a `disabled` badge in the plugin list. The plugin will still be considered as installed but disabled.

If you go into the plugin details page you won't be able to manage the plugin via the UI. This state requires you to take manual action to replace the plugin with a fully functioning version.

Skärmavbild 2021-09-14 kl  15 43 24

Skärmavbild 2021-09-14 kl  15 43 13

**Todo:**
- [x] Add some simple UI tests.
- [x] Set proper error messages for all the scenarios.

**Which issue(s) this PR fixes**:
Fixes #38719 

**Special notes for your reviewer**:
I still need to add good error messages on the details and tooltip of the error.

### CODE REVIEW

File: public/app/features/plugins/admin/api.ts [hunk 1]
```diff
@@ -59,6 +64,15 @@ async function getPlugin(slug: string): Promise<PluginDetails> {
   };
 }
 
+async function getPluginErrors(): Promise<PluginError[]> {
+  try {
+    const pluginErrors = await getBackendSrv().get(`${API_ROOT}/errors`);
+    return pluginErrors;
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/api.ts:70]: **Nit:** I think we can just return with the `await` and still being able to catch the errors

```suggestion
    return await getBackendSrv().get(`${API_ROOT}/errors`);;
```

File: public/app/features/plugins/admin/helpers.ts [hunk 2]
```diff
@@ -198,3 +214,10 @@ export const sortPlugins = (plugins: CatalogPlugin[], sortBy: Sorters) => {
 
   return plugins;
 };
+
+function groupByPluginId(errors: PluginError[]): Record<string, PluginError | undefined> {
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/helpers.ts:218]: **Nit:** maybe calling it something like `groupErrorsByPluginId` instead would reveal a bit more what it is doing

File: public/app/features/plugins/admin/components/PluginDetailsDisabledError.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          discovered that it was invalid. We can not guarantee the trustworthy of this plugin and have therefore
+          disabled it. We recommend you to reinstall the plugin to make sure you are running a verified version of this
+          plugin.
+        </p>
+      );
+    case PluginErrorCode.missingSignature:
+      return (
+        <p>
+          Grafana Labs checks each plugin to verify that it has a valid digital signature. While doing this, we
+          discovered that there are no signature for this plugin. We can not guarantee the trustworthy of this plugin
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/components/PluginDetailsDisabledError.tsx:56]: **Nit:** maybe a typo?
```suggestion
          discovered that there is no signature for this plugin. We can not guarantee the trustworthy of this plugin
```

**Review by leventebalogh — APPROVED** [review]
Nice! I like that you extracted to the badges to separate files 👏 

🚢 🇮🇹

File: public/app/features/plugins/admin/api.ts (hunk 1 above)
Comment (mckn) [reply public/app/features/plugins/admin/api.ts:70]: Hehe, not sure why I did that but great call!

**Review by achatterjee-grafana — APPROVED** [review]

**Pull request merged**